src/index.js

```diff
@@ -12,48 +12,41 @@ export default function loader(content) {
 
   const context = options.context || this.rootContext || (this.options && this.options.context);
 
-  let url = loaderUtils.interpolateName(this, options.name, {
+  const originalURL = loaderUtils.interpolateName(this, options.name, {
     context,
     content,
     regExp: options.regExp,
   });
 
-  let outputPath = '';
-
-  if (options.outputPath) {
-    // support functions as outputPath to generate them dynamically
-    outputPath = (
-      typeof options.outputPath === 'function' ? options.outputPath(url) : options.outputPath
-    );
-  }
+  let outputPath = (
+    typeof options.outputPath === 'function' ? options.outputPath(originalURL) : path.join((options.outputPath || '') + originalURL)
+  );
 
   if (options.useRelativePath) {
     const filePath = this.resourcePath;
-    const issuerContext = (this._module && this._module.issuer
-      && this._module.issuer.context) || context;
+    const issuerContext = context || (this._module && this._module.issuer
+      && this._module.issuer.context);
 
     const relativeUrl = issuerContext && path.relative(issuerContext, filePath).split(path.sep).join('/');
 
     const relativePath = relativeUrl && `${path.dirname(relativeUrl)}/`;
     // eslint-disable-next-line no-bitwise
     if (~relativePath.indexOf('../')) {
-      outputPath = path.posix.join(outputPath, relativePath, url);
+      outputPath = path.join(outputPath, relativePath, originalURL);
     } else {
-      outputPath = relativePath + url;
+      outputPath = path.normalize(relativePath + originalURL);
```

Comment:
This still concats 'malformed' `options.outputPath` fragments the wrong way ? `{ outpathPath: 'dist' }` => `dist6254824.svg`  (missing `'/'` (`dist/457438t3z58.svg`)) ?

Reply:
```js
path.normalize(path.join(relativePath, originalURL))
```

Reply:
Maybe already 'fixed' here https://github.com/webpack-contrib/file-loader/pull/250#discussion-diff-169404291R22

Reply:
@michael-ciniawsky let's add tests too :+1: